pulpcore/app/tasks/orphan.py

```
@@ -44,35 +49,68 @@ def orphan_cleanup(content_pks=None, orphan_protection_time=settings.ORPHAN_PROT
+            except ProtectedError:
+                # some orphan content might have been picked by another task running in parallel
+                # i.e. sync
+                for c in bulk_content:
+                    try:
+                        c.delete()
+                    except ProtectedError as e:
+                        log.debug(e)
+                        skipped_content += 1
+            progress_bar.increase_by(count)
+
+    if skipped_content:
+        msg = (
+            "{} orphaned content could not be deleted during this run and was skipped. "
+            "Re-run the task and/or consult the logs."
+        )
+        log.info(msg.format(skipped_content))
+        progress_bar.done -= skipped_content
+        progress_bar.save()
 
     # delete the artifacts that don't belong to any content
     artifacts = Artifact.objects.orphaned(orphan_protection_time)
 
+    skipped_artifact = 0
     with ProgressReport(
         message="Clean up orphan Artifacts",
         total=artifacts.count(),
         code="clean-up.artifacts",
     ) as progress_bar:
         for artifact in progress_bar.iter(artifacts.iterator()):
-            # we need to manually call delete() because it cleans up the file on the filesystem
-            artifact.delete()
+            try:
+                # we need to manually call delete() because it cleans up the file on the filesystem
+                artifact.delete()
+            except ProtectedError as e:
+                # some orphaned artifact might have been picked by another task running in parallel
+                # i.e. sync
+                log.debug(e)
+                skipped_artifact += 1
+
+    if skipped_artifact:
+        msg = (
+            "{} orphaned artifact(s) could not be deleted during this run and were skipped. "
+            "Re-run the task and/or consult the logs."
+        )
+        log.info(msg.format(skipped_artifact))
+
+        progress_bar.done -= skipped_artifact
```

I think I don't like this. Also it is inconsistent with how we report the deleted content.
Do we want to count deleted and skipped units separately?
Or have a total count and a skip count?

---

I did not want to introduce a separate report for skipped count. I could, but I don't like that. We would have in total 4 reports( 2 for success , 2 for skipped). This general pattern of adding reports( because of general report limitation)  for any other action except for `successful done` is not very useful. Something failed? add a report, something skipped? add report. As a result you have X separated reports instead of having 1 report that can count done, skipped,failed,etc.

To me, combined with the logs this is clear enough when i look at the report:
```
 {
      "message": "Clean up orphan Content",
      "code": "clean-up.content",
      "state": "completed",
      "total": 3,
      "done": 2,
      "suffix": null
    },
```
As i said, I can add additional reports, but for the mentioned reasons that would not be my preference. In addition, if on the user side there was some kind of script that was parsing this report, he'd need to adjust it to parse the newly added ones too. 
Le me know.

---

Adding a new counter should not break existing parsers (they should be built to ignore them.)
But I hear you. So your suggestion is: We set the total to the total number of processed units and done means "sucessfully done". And then the user needs to consult the state to see it is actually done. And the difference will be the failed units.
I believe we could do that, but i'd really not add the failed units to done to later subtract them again.

---

> Adding a new counter should not break existing parsers (they should be built to ignore them.) 

I agree but that means if ignored they would never know that 'something else happened'.
>But I hear you. So your suggestion is: We set the total to the total number of processed units and done means >"sucessfully done". And then the user needs to consult the state to see it is actually done. And the difference will be >the failed units. 

Yeah, that was  the idea behind. Unless we excplicity for some reason want the skipped/failed units to make visible in separate report, as, for example, there is such instance in rpm, plugin https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/tasks/synchronizing.py#L1278
> I believe we could do that, but i'd really not add the failed units to done to later subtract them again.

Yeah, I can keep separate variable counters for that to avoid the subtraction later on, just wanted to avoid more variable/counter to keep track off :unicorn: